File: ClientApp/src/app/models/project-task.ts
import { Basemodel } from './basemodel';
import { Workspace } from './workspace';
import { Project } from './project';
import { AppUser } from './appuser';

export class ProjectTask extends Basemodel {

    public name: string;
    public description: string;
    public status: ProjectTaskStatus;

    public project: Project;
    public projectId: number;

    public workspace: Workspace;
    public workspaceId: number;

    public assignee: AppUser;
    public assigneeId: string;
}

export enum ProjectTaskStatus {
    New = 0,
    Complete = 1,
    InProgress = 2,
    OnHold = 3,
    UnAssigned = 4,
    AwaitingClassification = 5,
    InActive = 6
}

File: Netptune.Models/Models/ProjectTask.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Netptune.Models.Enums;

namespace Netptune.Models.Models 
{
    public class ProjectTask : BaseModel
    {

        [Required]
        [StringLength(128)]
        public string Name { get; set; }

        [StringLength(1024)]
        public string Description { get; set; }

        public ProjectTaskStatus Status { get; set; }
        public double SortOrder { get; set; }

#region ForeignKeys

        [ForeignKey ("Assignee")]
        public string AssigneeId { get; set; }

        [Required]
        [ForeignKey ("Project")]
        public int? ProjectId { get; set; }

        [Required]
        [ForeignKey ("Workspace")]
        public int? WorkspaceId { get; set; }

#endregion

#region ForeignKeys

        public virtual AppUser Assignee { get; set; }
        public virtual Project Project { get; set; }
        public virtual Workspace Workspace { get; set; }

#endregion
    }
}
